src/engines/algorithms_config.py

```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, List
+
+from algorithms import Algorithm
+from algorithms.associations import APriori
+from algorithms.classification import ExtraTrees
+from algorithms.clustering import GMM, KMeans
+from widgets.options_widgets import (
+    AlgorithmOptions,
+    AssociationRulesOptions,
+    ExtraTreesOptions,
+    GMMOptions,
+    KMeansOptions,
+)
+from widgets.results_widgets import (
+    AlgorithmResultsWidget,
+    APrioriResultsWidget,
+    ExtraTreesResultsWidget,
+    GMMResultsWidget,
+    KMeansResultsWidget,
+)
+from widgets.steps_widgets import (
+    AlgorithmStepsVisualization,
+    APrioriStepsVisualization,
+    ExtraTreesStepsVisualization,
+    GMMStepsVisualization,
+    KMeansStepsVisualization,
+)
+
+
+class AlgorithmTechniques(Enum):
+    CLUSTERING = "clustering"
+    ASSOCIATIONS = "associations"
+    CLASSIFICATION = "classification"
+
+    @classmethod
+    def list(cls) -> List[str]:
+        return list(map(lambda e: e.value, cls))
+
+
+@dataclass
+class AlgorithmConfig:
+    algorithm: Algorithm.__class__
+    options: AlgorithmOptions.__class__
+    steps_visualization: AlgorithmStepsVisualization.__class__
+    result_widget: AlgorithmResultsWidget.__class__
```

Comment:
```suggestion
    algorithm: Type[Algorithm]
    options: Type[AlgorithmOptions]
    steps_visualization: Type[AlgorithmStepsVisualization]
    result_widget: Type[AlgorithmResultsWidget]
```
`Type` from `typing` module. looks better imo